### PR TITLE
Add continous value (plDDT and PPL) support for Curriculum Learning

### PR DESCRIPTION
**Pull request by Leo-T-Zang**
1. Enhanced the curriculum learning strategy with continuous values (like ppl and plddt) into categories. We notice it actually supports contnuouse values.
2. Testing script:
   - Leveraged the existing testing script of length to validate the new PPL one.

Both plDDT and PPL values should be pre-computed and saved within dataset.

### CODE REVIEW

**Review by pascalnotin — COMMENTED** [review]
Hi @Leo-T-Zang - nice work! 
Had a couple questions/comments before approving:
1. What is the role of the new fields in the yaml files (eg., ppl_category, plddt_category)? They do not seem to be used in the rest of the code
2. We probably want to generate a sequence of amino acids [here](https://github.com/OpenBioML/protein-lm-scaling/pull/58/files#diff-e5d59066ce6d535ba1b2073e1205ff2d464483cd8fda4feb0d839aa90ab7d15cR100) instead :)
3. I would add two small scripts in dataset.py to compute the perplexity metric based on ESM2, and the pLDDT based on ESMFold - and have a separate test to ensure they return correct values. In practice we will then run these scripts on the training datasets to pre-compute the values necessary to the CL scheme you implemented

**Comment by pascalnotin** [comment]
Another comment as Im re-reading the test routine: we should check that the variable of interest (eg., ppl) is ordered throughout training across mini batches (not within a mini batch).

**Comment by talkhanz** [comment]
> Another comment as Im re-reading the test routine: we should check that the variable of interest (eg., ppl) is ordered throughout training across mini batches (not within a mini batch).

I can make this change in a subsequent PR

**Comment by pascalnotin** [comment]
That's a good point. I assumed things would be different for sequence length vs other CL strategies bc the length of input can be straightforwardly computed vs other strategies can be more involved (ie., calls for separate models such as Tranception to compute the perplexity or ESMfold to compute the plddt). But these more cpx strategies can also be handled via the batch_set function (as per your point S2) then it's probably easier to go with that

**Review by pascalnotin — APPROVED** [review]
Merging this PR based on pre-xmas discussion: we will pre-compute values to be used for the Curriculum Learning and store them together with the overarching cluster mapping file. This mapping file will thus contain cluster name, cluster representative sequence, pointer to file location on disk where all sequences in that cluster are stored, plDDT and PPL for the cluster representative sequence computed with a pretrained model (eg., ESM2 or Tranception).

**Comment by pascalnotin** [comment]
@Leo-T-Zang @talkhanz @jamaliki for reference ^^
